src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -63,35 +69,60 @@ case class Face(normalVec: (Double,Double,Double), distance: Double, phi: Double
     (math.abs(c + inputFace.c) < tolerance) &&
     (math.abs(d + inputFace.d) < tolerance)
   }
+
+  override def equals(obj: Any): Boolean = {
+    obj match {
+      case f: Face =>
+        this.a == f.a && this.b == f.b && this.c == f.c &&
+        this.distance == f.distance && this.phi == f.phi &&
+        this.cohesion == f.cohesion &&
+        this.processorJoint == f.processorJoint
+
+      case _ => false
+    }
+  }
+
+  override def hashCode: Int =
+    new HashCodeBuilder()
+        .append(a)
+        .append(b)
+        .append(c)
+        .append(distance)
+        .append(phi)
+        .append(cohesion)
+        .append(processorJoint)
+        .toHashCode
 }
 
 object Block {
   /**
     * Calculates the rotation matrix to rotate the input plane (specified by its normal)
     * to the desired orientation (specified by the desired normal)
-    * @param n_current: Current normal of plane
-    * @param n_desired: Desired new normal
+    *
+    * @param nCurrent: Current normal of plane
+    * @param nDesired: Desired new normal
     * @return 3*3 rotation matrix
     */
-  def rotationMatrix(n_current: (Double, Double, Double), n_desired: (Double, Double, Double)):
-      DenseMatrix[Double] = {
-    val n_c = DenseVector[Double](n_current._1, n_current._2, n_current._3)
-    val n_d = DenseVector[Double](n_desired._1, n_desired._2, n_desired._3)
+  def rotationMatrix(nCurrent: Array[Double], nDesired: Array[Double]): DenseMatrix[Double] = {
+    assert(nCurrent.length == 3 && nDesired.length == 3)
+
+    val nC = DenseVector[Double](nCurrent(0), nCurrent(1), nCurrent(2))
+    val nD = DenseVector[Double](nDesired(0), nDesired(1), nDesired(2))
     // Check if vectors are parallel
-    if (math.abs(linalg.norm(linalg.cross(n_c,n_d))) > NumericUtils.EPSILON) {
-      val v = linalg.cross(n_c, n_d)
+    if (math.abs(linalg.norm(linalg.cross(nC, nD))) > NumericUtils.EPSILON) {
+      val v = linalg.cross(nC, nD)
       val s = linalg.norm(v)
-      val c = n_c dot n_d
+      val c = nC dot nD
 
-      val A_skew = DenseMatrix.zeros[Double](3,3)
-      A_skew(0,1) = -v(2)
-      A_skew(0,2) = v(1)
-      A_skew(1,0) = v(2)
-      A_skew(1,2) = -v(0)
-      A_skew(2,0) = -v(1)
-      A_skew(2,1) = v(0)
+      val ASkew = DenseMatrix.zeros[Double](3,3)
+      ASkew(0,1) = -v(2)
+      ASkew(0,2) = v(1)
+      ASkew(1,0) = v(2)
+      ASkew(1,2) = -v(0)
+      ASkew(2,0) = -v(1)
+      ASkew(2,1) = v(0)
 
-      DenseMatrix.eye[Double](3) + A_skew + (A_skew * A_skew) * (1-c)/(s*s)
+      DenseMatrix.eye[Double](3) + ASkew + (ASkew * ASkew) * (1-c)/(s*s)
```

Comment:
Well, the standard in Java and Scala is to always use camel case. Technically, all variable names should also start with lower case letters, but since we're dealing with matrices here, I think this is okay.